src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -247,6 +250,210 @@ public static RecordValue Execute()
+            }
+
+            // Must have "Execute" method. 
+            public static RecordValue Execute()
+            {
+                var recordType = RecordType.Empty()
+                    .Add(new NamedFormulaType("num1", FormulaType.Number));
+                var val = FormulaValue.NewRecordFromFields(recordType, new NamedValue("num1", FormulaValue.New(1)));
+                return val;
+            }
+        }
+
+        [Fact]
+        public void CustomFunctionRecordTest()
+        {
+            CustomFunctionRecord(new TestSuccessRecordCustomFunction(), "SuccessRecordTest", "({x: 2, y: 5})", true);
+
+            CustomFunctionRecord(new TestMissingFieldRecordCustomFunction(), "MissingFieldRecordTest", "({x: 2})", true);
+
+            CustomFunctionRecord(new TestSuccessCoerceRecordCustomFunction(), "SuccessCoerceRecordTest", "({x: \"2\", y: \"5\"})", true);
+
+            CustomFunctionRecord(new TestSuccessNestedRecordCustomFunction(), "SuccessNestedRecordTest", "({x: {x : 2}})", true);
+
+            CustomFunctionRecord(new TestExtraFieldRecordCustomFunction(), "ExtraFieldRecordTest", "({x: 2, y: 5, a: 8})", false);
+
+            CustomFunctionRecord(new TestNotExistRecordCustomFunction(), "NotExistRecordTest", "({a: 8})", false);
+            
+            CustomFunctionRecord(new TestNestedRecordCustomFunction(), "NestedRecordTest", "({x: {x : \"2\"}})", false);
+        }
+
+        private void CustomFunctionRecord(ReflectionFunction reflectionFunction, string functionName, string inputRecord,  bool isSuccess)
+        {
+            var config = new PowerFxConfig();
+            config.AddFunction(reflectionFunction);
+            
+            var engine = new RecalcEngine(config);
+            var func = engine.GetFunctionsByName(functionName);
+            Assert.NotNull(func);
+
+            var check = engine.Check(functionName + inputRecord);
+
+            FormulaValue result = null; 
+            ConfiguredTaskAwaitable<FormulaValue> resultAsync;
+            string errorMsg = string.Empty;
+
+            try
+            {
+                result = engine.Eval(functionName + inputRecord);
+
+                using var cts = new CancellationTokenSource();
+                resultAsync = engine.EvalAsync(functionName + inputRecord, cts.Token).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                errorMsg = ex.ToString();
+            }
+
+            if (isSuccess)
+            {
+                Assert.True(check.IsSuccess);                
+                Assert.IsNotType<ErrorValue>(result);
+                Assert.IsNotType<ErrorValue>(resultAsync);
+            }
+            else
+            {
+                Assert.False(check.IsSuccess);
+                Assert.Contains("invalid arguments", errorMsg);
+            }
+        }
+
+        // Must have "Function" suffix. 
+        private class TestSuccessRecordCustomFunction : ReflectionFunction
```

Comment:
>TestSuccessRecordCustomFunction [](http://example.com/codeflow?start=22&length=31)

Suggestion: instead of having multiple functions, have a single custom (reflection) function that takes multiple arguments - and in the test pass blank value to the arguments that don't apply to the test.

Reply:
Good suggestion. Update the code to remove duplicated.